src/lib/yang/schema.lua

```diff
@@ -564,6 +570,26 @@ local primitive_types = set(
    'binary', 'bits', 'boolean', 'decimal64', 'empty', 'enumeration',
    'identityref', 'instance-identifier', 'leafref', 'string', 'union')
 
+-- Inherits config attributes from parents
+local function inherit_config(schema, env)
+   if env == nil then env = {env={}} end
+
+   if schema.config ~= nil then
+      env.config = schema.config
+   elseif env.env.config ~= nil then
+      schema = shallow_copy(schema)
+      schema.config = env.env.config
+   end
+
+   if schema.body then
```

Comment:
does it? We shallow_copy if on the schema before we ever modify it. I.e. we either return schema completely unmodified or, in the case we modify it, we shallow copy it first and return that. 

The reason I'm doing it this way round is there are plenty of places where config isn't set or nothing needs to be changed so shallow copying is a waste of time, it seems better to only copy if mutating.

WDYT?